docs/DeveloperGuide.md

```diff
@@ -419,6 +426,9 @@ of a `StatisticsCommand`:
     * Easy, straightforward to implement.
   * Cons:
     * Violates the law of demeter to a large extent.
+    
+Alternative 1 was chosen because we want to maintain the testability of our code. Testing would be harder with
+alternative 2 since everything is done within `StatisticsCommand#execute()`.
```

Comment:
Perhaps it would be good to link with abstraction as the link may not be very obvious.